modules/cloud_logs/main.tf

```diff
@@ -33,7 +33,11 @@ resource "ibm_resource_tag" "cloud_logs_tag" {
 # Get Cloud Account ID
 ##############################################################################
 
-data "ibm_iam_account_settings" "iam_account_settings" {
+# If logs or metrics data is enabled, parse details from it
+module "cos_crn_parser" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.1.0"
+  crn     = var.data_storage.logs_data.enabled ? var.data_storage.logs_data.bucket_crn : (var.data_storage.metrics_data.enabled ? var.data_storage.metrics_data.bucket_crn : null)
```

Comment:
if a user didn't pass any buckets, then this will fail as it will try to pass `null` for the required `crn` input. You need to add a count around this module to ensure its only executed if a bucket is being passed for logs or metrics